cypress/support/custom_commands/login.helper.js

```diff
@@ -0,0 +1,193 @@
+		case 'student2_nbc':
+			return ['STUDENT_2_NBC_EMAIL', 'STUDENT_2_NBC_PASSWORD']
+		case 'admin1_nbc':
+			return ['ADMIN_1_NBC_EMAIL', 'ADMIN_1_NBC_PASSWORD']
+		case 'admin1_dbc':
+			return ['ADMIN_1_DBC_EMAIL', 'ADMIN_1_DBC_PASSWORD']
+		case 'student_extern_dbc':
+			return ['STUDENT_DBC_EXTERN', 'STUDENT_DBC_EXTERN_PASSWORD']
+		default:
+			return [null, null]
+	}
+}
+
+const performExternalLogin = () => {
+	cy.request('GET', oauth_url).then(resp => {
+		cy.intercept(resp.requestHeaders.referer).as('oauth_url')
+		cy.visit(resp.requestHeaders.referer).then(window => {
+			expect(window.location.pathname).to.include('/Account/Login')
+			cy.get('@oauth_url').then(resp => {
+				expect(resp.response.statusCode).to.equal(308)
+				cy.get(externalUsernameInputFieldElement).should('be.visible')
+				cy.get(externalUsernameInputFieldElement).type(env[userEmail], {
+					log: false
+				})
+				cy.get(externalPasswordInputFieldElement)
+					.type(env[userPassword], { log: false })
+					.type('{enter}')
+			})
+		})
+	})
+}
+
+const fillLoginForm = (username, password) => {
+	return (
+		cy.get(emailInputFieldElement).type(username, { log: true }),
+		cy.get(passwordInputFieldElement).type(password, { log: true }),
+		cy.get(submitButton).click()
+	)
+}
+
+const shuffleString = str => {
+	const characters = str.split('')
+	let currentIndex = characters.length
+	let randomIndex
+
+	while (currentIndex !== 0) {
+		randomIndex = Math.floor(Math.random() * currentIndex)
+		currentIndex--
+		;[characters[currentIndex], characters[randomIndex]] = [
+			characters[randomIndex],
+			characters[currentIndex]
+		]
+	}
+
+	return characters.join('')
+}
+const generateStrongPassword = length => {
+	const lowercase = 'abcdefghijklmnopqrstuvwxyz'
+	const uppercase = lowercase.toUpperCase()
+	const numbers = '0123456789'
+	const symbols = '!@#$%^&*'
+
+	const allChars = lowercase + uppercase + numbers + symbols
+
+	let password = ''
+	for (let i = 0; i < length; i++) {
+		const charSet = allChars[Math.floor(Math.random() * allChars.length)]
+		const randomIndex = Math.floor(Math.random() * charSet.length)
+		password += charSet[randomIndex]
+	}
+
+	return shuffleString(password)
+}
+
+const studentFirstLogin = () => {
+	const newPassword = generateStrongPassword(12)
+	Cypress.env('password', newPassword)
+	cy.log('================================')
+	cy.log(env['username'])
+	cy.log(env['password'])
+	cy.get(studentAgeSelectRadioBtn).check()
+	cy.get(nextButtonAfterAgeSelection).click()
+	cy.get(nextButtonOnFirstLoginPages).click()
+	cy.get(nextButtonOnFirstLoginPages).click()
+	cy.get(datePickerSelectorForDOB).type('21').type('03').type('2000')
+	cy.get(nextButtonOnFirstLoginPages).click()
+	cy.get(studentUpdatePassword).type(env['password'])
+	cy.get(studentConfirmPassword).type(env['password'])
+	cy.get(nextButtonOnFirstLoginPages).click()
+	cy.get(skipToDashboardButtonOnFirstLoginPage).click()
+}
+
+const otherUsersFirstLogin = () => {
```

Comment:
otherUsersFirstLogin should be more specific if its for admin, teacher etc.

Reply:
Done